lib/ClangImporter/SwiftLookupTable.h

```diff
@@ -196,6 +196,8 @@ class EffectiveClangContext {
       DC = omDecl->getCanonicalDecl();
     } else if (auto fDecl = dyn_cast<clang::FunctionDecl>(dc)) {
       DC = fDecl->getCanonicalDecl();
+    } else if (auto nsDecl = dyn_cast<clang::NamespaceDecl>(dc)) {
+      DC = nsDecl->getCanonicalDecl();
```

Comment:
This is eventually going to need more work, since the same namespace can be added to from multiple modules and it will basically hop around based on what you have loaded. We should talk about this specifically later—I'm inclined to think we should put all namespaces in a dummy module rather than try to tie them to particular modules.

Reply:
I'm interested to hear what you think about namespaces once they're imported into swift. I will note, however, that this particular line is for finding the decl context of a particular c++ decl. We then later decide how this is imported into swift. I think the least surprising thing when it comes to that would be to import them in a similar fashion to how nominal types and their extensions are currently handled in swift. I believe there is an explicit extension decl that exposes the decls defined in the extension.

Reply:
There is, but there also has to be a nominal to extend, and that nominal has to be in some module.